src/main/java/de/rabea/request/HttpVerb.java

```diff
@@ -15,6 +15,12 @@ public static HttpVerb convert(String word) {
                 return verb;
             }
         }
-        return null;
+        throw new UnknownHttpVerbException("Not a valid Http verb");
```

Comment:
I would move the message into the Exception itself, unless you intend to pass different messages in different occasions...
Plus: have you considered the "apologies and complaints" style for exceptions? "Apologies" when the client could not have done anything to resolve the situation, "Complaint" when the client passes bad data or some other violation...